www/docs/configuration/providers.md

```diff
@@ -20,42 +29,73 @@ NextAuth.js is designed to work with any OAuth service, it supports OAuth 1.0, 1
 
 2. The redirect URI should follow this format:
-  ```
-  [origin]/api/auth/callback/[provider]
-  ```
-  For example, Twitter on `localhost` this would be:
-  ```
-  http://localhost:3000/api/auth/callback/twitter
-  ```
+
+```
+[origin]/api/auth/callback/[provider]
+```
+
+For example, Twitter on `localhost` this would be:
+
+```
+http://localhost:3000/api/auth/callback/twitter
+```
+
 3. Create a `.env` file at the root of your project and add the client ID and client secret. For Twitter this would be:
 
-  ```
-  TWITTER_ID=YOUR_TWITTER_CLIENT_ID
-  TWITTER_SECRET=YOUR_TWITTER_CLIENT_SECRET
-  ```
+```
+TWITTER_ID=YOUR_TWITTER_CLIENT_ID
+TWITTER_SECRET=YOUR_TWITTER_CLIENT_SECRET
+```
 
 4. Now you can add the provider settings to the NextAuth options object. You can add as many OAuth providers as you like, as you can see `providers` is an array.
 
-  ```js title="pages/api/auth/[...nextauth].js"
-  import Providers from `next-auth/providers`
-  ...
-  providers: [
-    Providers.Twitter({
-      clientId: process.env.TWITTER_ID,
-      clientSecret: process.env.TWITTER_SECRET
-    })
-  ],
-  ...
-  ```
-5. Once a provider has been setup, you can sign in at the following URL: `[origin]/api/auth/signin`. This is an unbranded auto-generated page with all the configured providers.   
+```js title="pages/api/auth/[...nextauth].js"
+import Providers from `next-auth/providers`
+...
+providers: [
+  Providers.Twitter({
+    clientId: process.env.TWITTER_ID,
+    clientSecret: process.env.TWITTER_SECRET
+  })
+],
+...
+```
+
+5. Once a provider has been setup, you can sign in at the following URL: `[origin]/api/auth/signin`. This is an unbranded auto-generated page with all the configured providers.
 
 <Image src="/img/signin.png" alt="Signin Screenshot" />
 
+### Options
+
+|        Name         |                           Description                            |               Type                | Required |
+| :-----------------: | :--------------------------------------------------------------: | :-------------------------------: | :------: |
+|         id          |                    Unique ID for the provider                    |             `string`              |   Yes    |
+|        name         |                Descriptive name for the provider                 |             `string`              |   Yes    |
+|        type         |              Type of provider, in this case `oauth`              |              `oauth`              |   Yes    |
+|       version       |            OAuth version (e.g. '1.0', '1.0a', '2.0')             |             `string`              |   Yes    |
+|        scope        |          OAuth access scopes (expects array or string)           |      `string` or `string[]`       |   Yes    |
+|       params        |             Additional authorization URL parameters              |     `{ grant_type: string }`      |   Yes    |
```

Comment:
I think the description is wrong on this one (that said, the name is also too generic). I believe this is used to send extra params to the `/token` endpoint when getting an access_token from the OAuth provider. The type is not limited to `grant_type` either.

We do some magic :rainbow: with the params (open the link and scroll, there is more than in this snippet.): https://github.com/nextauthjs/next-auth/blob/dfe4620056b6ba3f72d1dd5ab166efb7f29db5c7/src/server/lib/oauth/client.js#L102-L115

We do some provider-specific things here that shouldn't be in the core, and I hope that #1846 will clear this up because right now it is quite confusing and not flexible enough.